getMultiStarts.m

```diff
@@ -117,61 +117,68 @@
 switch options.proposal
     case 'latin hypercube'
         % Sampling from latin hypercube
         par0 = [parameters.guess,...
-                bsxfun(@plus,parameters.min,bsxfun(@times,parameters.max - parameters.min,...
-                             lhsdesign(options.n_starts - size(parameters.guess,2),parameters.number,'smooth','off')'))];
+            bsxfun(@plus,parameters.min,bsxfun(@times,parameters.max - parameters.min,...
+            lhsdesign(options.n_starts - size(parameters.guess,2),parameters.number,'smooth','off')'))];
     case 'uniform'
         % Sampling from uniform distribution
         par0 = [parameters.guess,...
-                bsxfun(@plus,parameters.min,bsxfun(@times,parameters.max - parameters.min,...
-                             rand(parameters.number,options.n_starts - size(parameters.guess,2))))];
+            bsxfun(@plus,parameters.min,bsxfun(@times,parameters.max - parameters.min,...
+            rand(parameters.number,options.n_starts - size(parameters.guess,2))))];
     case 'user-supplied'
         % Sampling from user-supplied function
         if (~isfield(parameters, 'init_fun') || isempty(parameters.init_fun))
             if size(parameters.guess,2) < options.n_starts
                 error('You did not define an initial function and do not provide enough starting points in parameters.guess. Aborting.');
             else
-                par0 = [parameters.guess(:,1:options.n_starts)];
+                par0 = parameters.guess(:,1:options.n_starts);
             end
         else
             par0 = [parameters.guess,...
-            parameters.init_fun(parameters.guess,parameters.min,parameters.max,options.n_starts - size(parameters.guess,2))];
-        end 
+                parameters.init_fun(parameters.guess,parameters.min,parameters.max,options.n_starts - size(parameters.guess,2))];
+        end
 end
+% Correct for fixed parameters
+par0 = par0(freePars,:);
 parameters.MS.n_starts = options.n_starts;
-parameters.MS.par0 = par0(:,options.start_index);
+parameters.MS.par0 = nan(parameters.number,max(options.start_index));
+parameters.MS.par0(freePars,:) = par0(:,options.start_index);
+parameters.MS.par0(options.fixedParameters,:) = options.fixedParameterValues;
 
 %% Preparation of folder
 if or(options.save,options.tempsave)
     if(~exist(fullfile(pwd,options.foldername),'dir'))
         mkdir(fullfile(pwd,options.foldername))
     end
     % only save the init mat for the first start index, not every one if they are called seperately
-    if(and(options.save,~isempty(find(options.start_index==1))))
+    if(and(options.save,~isempty(find(options.start_index == 1, 1))))
         save([options.foldername '/init'],'parameters','-v7.3');
     end
 end
 
 %% Initialization
-if strcmp(options.localOptimizer, 'fmincon') || strcmp(options.localOptimizer, 'pswarm')
+if (strcmp(options.localOptimizer, 'fmincon') || strcmp(options.localOptimizer, 'pswarm') || strcmp(options.localOptimizer, 'lsqnonlin'))
```

Comment:
again check if the respective field exists? Only make special cases if for a particular optimizer the options does a different thing (which hopefully does not happen)